sdk/src/entrypoint.rs

```diff
@@ -1,48 +1,10 @@
-//! @brief Solana Rust-based BPF program entry point supported by the latest
-//! BPFLoader.  For more information see './bpf_loader.rs'
+pub use solana_program_sdk::entrypoint::*;
 
-extern crate alloc;
-use crate::{account_info::AccountInfo, program_error::ProgramError, pubkey::Pubkey};
-use alloc::vec::Vec;
-use std::{
-    alloc::Layout,
-    cell::RefCell,
-    mem::{align_of, size_of},
-    ptr::null_mut,
-    rc::Rc,
-    // Hide Result from bindgen gets confused about generics in non-generic type declarations
-    result::Result as ResultGeneric,
-    slice::{from_raw_parts, from_raw_parts_mut},
-};
-
-pub type ProgramResult = ResultGeneric<(), ProgramError>;
-
-/// User implemented function to process an instruction
-///
-/// program_id: Program ID of the currently executing program accounts: Accounts
-/// passed as part of the instruction instruction_data: Instruction data
-pub type ProcessInstruction =
-    fn(program_id: &Pubkey, accounts: &[AccountInfo], instruction_data: &[u8]) -> ProgramResult;
-
-/// Programs indicate success with a return value of 0
-pub const SUCCESS: u64 = 0;
-
-/// Start address of the memory region used for program heap.
-pub const HEAP_START_ADDRESS: usize = 0x300000000;
-/// Length of the heap memory region used for program heap.
-pub const HEAP_LENGTH: usize = 32 * 1024;
-
-/// Declare the entry point of the program and use the default local heap
-/// implementation
-///
-/// Deserialize the program input arguments and call the user defined
-/// `process_instruction` function. Users must call this macro otherwise an
-/// entry point for their program will not be created.
-///
-/// If the program defines the feature `custom-heap` then the default heap
-/// implementation will not be included and the program is free to implement
-/// their own `#[global_allocator]`
 #[macro_export]
+#[deprecated(
+    since = "1.4.3",
+    note = "use solana_program_sdk::entrypoint::entrypoint instead"
+)]
 macro_rules! entrypoint {
```

Comment:
This sucks a little.  I had to keep an `entrypoint` macro in here to avoid breaking users

Reply:
Which users are you worried about, wouldn't anyone picking this sdk need to start using solana-program-sdk anyway?

Reply:
> wouldn't anyone picking this sdk need to start using solana-program-sdk anyway?

Currently you can just upgrade solana-sdk and it'll work (with some deprecated compiler warning).  I expect this will be the normal upgrade path for downstream:
1. Next year upgrade to a new solana-sdk
2. Notice the deprecated compiler warnings
3. Switch to solana-program-sdk in 5 years